modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -820,6 +820,117 @@ public String getSucessorPredecessorSubprocessList(String resourcePath){
         return resourceString;
     }
 
+    public String addSubprocess(String subprocessDetails){
+        try{
+            RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+
+            if(registryService != null){
+                UserRegistry reg = registryService.getGovernanceSystemRegistry();
+
+                JSONObject processInfo = new JSONObject(subprocessDetails);
+                String processName = processInfo.getString("processName");
+                String processVersion = processInfo.getString("processVersion");
+                JSONArray subprocessList = processInfo.getJSONArray("subprocessList");
+
+                String processAssetPath = "processes/" + processName + "/" + processVersion;
+                Resource resource = reg.get(processAssetPath);
+                String processContent = new String((byte[]) resource.getContent());
+                Document doc = stringToXML(processContent);
+
+                if(subprocessList.length() != 0){
+                    Element rootElement = doc.getDocumentElement();
+                    for(int i = 0 ; i < subprocessList.length() ; i++){
+                        Element successorElement = append(doc, rootElement, "subprocess", mns);
+                        appendText(doc, successorElement, "name", mns, subprocessList.getJSONObject(i).getString("name"));
+                        appendText(doc, successorElement, "path", mns, subprocessList.getJSONObject(i).getString("path"));
+                        appendText(doc, successorElement, "id", mns, subprocessList.getJSONObject(i).getString("id"));
+                    }
+                    String newProcessContent = xmlToString(doc);
+                    resource.setContent(newProcessContent);
+                    reg.put(processAssetPath, resource);
+                }
+            }
+
+        }catch (Exception e){
+            log.error(e);
```

Comment:
Better to log the error with a custom message to uniquely identify the source of the error. 
log.error("Error when adding sub process", e);
Also whenever possible better to catch the specific exception rather than catching the super object Exception.